Mathlib/Algebra/Order/Ring/Star.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2023 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Algebra.Order.Ring.Defs
+import Mathlib.Algebra.Star.Order
+
+/-!
+# Commutative star-ordered rings are ordered rings
+
+A noncommutative star-ordered ring is generally not an ordered ring. Indeed, in a star-ordered
+ring, nonnegative elements are self-adjoint, but the product of self-adjoint elements is
+self-adjoint if and only if they commute. Therefore, a necessary condition for a star-ordered ring
+to be an ordered ring is that all nonnegative elements commute.
+
+In this file we prove a kind of converse: a *commutative* star-ordered ring is an ordered ring.
+-/
+
+namespace StarOrderedRing
+
+/- This example shows that nonnegative elements in a ordered semiring which is also star-ordered
+must commute. -/
+example {R : Type*} [OrderedSemiring R] [StarOrderedRing R] {x y : R} (hx : 0 ≤ x) (hy : 0 ≤ y) :
+    x * y = y * x := by
```

Comment:
Can you add to this comment an explanation of why you think this would be a pointless lemma?